docs/source/guides/mod-config.md

```diff
@@ -1,166 +1,323 @@
 
-!!! example "**MyMod\\main.lua**"
-	```lua
+```lua title="My Awesome Mod/main.lua"
 	local settings = template:createSideBarPage({
 		label = "Settings",
 		-- This description will be displayed at the right panel if no component is moused over.
+		-- (The `\z` character lets you define a string across multiple lines.)
 		description =
-			"My Awesome Mod\n" ..
-			"Version 1.0.0\n" ..
-			"\n" ..
-			"The mod is very awesome. And this is its awesome description.\n",
+			"My Awesome Mod\n\z
+			Version 1.0.0\n\z
+			\n\z
+			The mod is very awesome. And this is its awesome description.\n",
 	})
 
 	settings:createYesNoButton({
 		label = "Enable Mod",
 		-- This description will be displayed at the right panel if the button is moused over.
 		description =
-			"If this setting is enabled, the mod is enabled.\n" ..
-			"\n" ..
-			"If this setting is disabled, the mod is disabled.\n" ..
-			"\n" ..
-			"Default: yes",
+			"If this setting is enabled, the mod is enabled.\n\z
+			\n\z
+			If this setting is disabled, the mod is disabled."
 		variable = mwse.mcm:createTableVariable({ id = "enabled", table = config }),
 	})
-	```
+```
+
+### Displaying Default Values of Settings
+As your config menu grows, you may want to include some information about what the default values for various settings are. 
+Fortunately, the MCM comes with a very straight-forward solution to this problem:
+```lua hl_lines="4 5"
+local template = mwse.mcm.createTemplate({ 
+	name = "My Awesome Mod", 
+	config = myConfig,
+	defaultConfig = myDefaultConfig,
+	showDefaultSetting = true,
+})
+```
 
-### Key Binder
+This will result in the default values of settings automatically being displayed underneath their descriptions, provided those settings are in a `SideBarPage`.
 
-If your mod adds any hotkey function, a way to rebind the hotkey in the mod config menu is a must.
+### Separate config.lua
 
-!!! example "**MyMod\\main.lua**"
-	```lua
-	settings:createKeyBinder({
-		label = "Assign Keybind",
-		description = "Assign a new keybind to perform awesome tasks.",
-		variable = mwse.mcm.createTableVariable{ id = "keybind", table = config },
-		allowCombinations = true,
-	})
-	```
+As your default config gets long, you might want to create a separate Lua file for it.
+
+```lua linenums="1" title="My Awesome Mod/config.lua"
+local defaultConfig = {
+	enabled = true,
+	keybind = {
+		keyCode = tes3.scanCode.k,
+		isShiftDown = false,
+		isAltDown = false,
+		isControlDown = true,
+	}, -- Ctrl + K
+	nonEdible = {
+		["ingred_diamond_01"] = true,
+		["ingred_scrap_metal_01"] = true,
+	}
+}
+local config = mwse.loadConfig("My Awesome Mod", defaultConfig)
+-- Return both the `config` and the `defaultConfig`, so that the MCM can properly
+-- display default values.
+return config, defaultConfig
```

Comment:
Oh oops. I should have tested this part first, I just assumed it worked and was planning on testing everything a bit later.

How do you think I should fix this? Because storing the config in a separate file is important, but it would also be nice to give the MCM access to the default config. I'm currently storing the config and default config in two separate files, using the structure
```
My Mod/config/init.lua
My Mod/config/default.lua
```
But I worry that advising people make two separate files and a new folder might be a bit much in an introductory guide. Especially if it involves explaining that `config/init.lua` means you can still write `require("my mod.config")`.

Am I overthinking this? Maybe I could just advise having a `defaultConfig.lua` file and a `config.lua` file?

Reply:
What if there is only `config.lua` returning `{ config = config, defaultConfig = defaultConfig }`?

Reply:
I think that this might be a good solution. The only thing I worry about is that in these beginner guides, the 
```lua
local keyName1 = someValue1
local keyName2 = someValue2
local myTable = { keyName1 = keyName1, keyName2 = keyName2 }
```
syntax might be confusing. I think it's the kind of thing that's not confusing once you're used to it, but I've seen a few people get confused it when first starting out.

How would you feel about something like `config = myConfig` and `defaultConfig = myDefaultConfig`?
Another option would be

```lua
local configs = {}

configs.defaultConfig =  { ... }

configs.loadedConfig = mwse.loadConfig(fileName, configs.defaultConfig)

return configs
```

But maybe `configs.default` and `configs.loaded` would be better?

Reply:
In general I'm okay with any of your proposals. I'll give one more if you rename your `configs` table to `this` you are left with `config` for the config proper:
```lua
local this= {}

this.defaultConfig =  { ... }

this.config = mwse.loadConfig(fileName, this.defaultConfig)

return this
```

Reply:
I think this is also a good idea. I personally found it a bit confusing when importing things later because it could lead to things like 
```lua
local configs = require("My Awesome Mod.configs")
local config = configs.config
local defaultConfig = configs.defaultConfig
```
and I (personally) found the `configs` and `config` distinction a bit confusing. But that could probably be resolved with either using different variable names or just doing 

```lua
local config = require("My Awesome Mod.configs").config
local defaultConfig = require("My Awesome Mod.configs").defaultConfig
```
I'm happy with things the way they are but I can change things to use this approach if you would prefer.

Reply:
It's fine now.